src/runtime/composables/useSession.ts

```diff
@@ -204,6 +205,17 @@ const getSession = async (getSessionOptions?: GetSessionOptions) => {
     onResponse: ({ response }) => {
       const sessionData = response._data
 
+      // Add any new cookie to the server-side event for it to be present on the app-side after
+      // initial load, see sidebase/nuxt-auth/issues/200 for more information.
+      if (process.server) {
+        if (response.headers.get('set-cookie') && nuxt.ssrContext) {
+          const setCookieValue = response.headers.get('set-cookie') || ''
+          if (setCookieValue) {
+            appendHeader(nuxt.ssrContext.event, 'set-cookie', setCookieValue)
+          }
+        }
+      }
```

Comment:
I think we can improve this a bit:
```suggestion
      // Add any new cookie to the server-side event for it to be present on the app-side after
      // initial load, see sidebase/nuxt-auth/issues/200 for more information.
      if (process.server) {
        const setCookieValue = response.headers.get('set-cookie')
        if (setCookieValue && nuxt.ssrContext) {
          appendHeader(nuxt.ssrContext.event, 'set-cookie', setCookieValue)
        }
      }
```